libraries/ESP8266WiFi/src/ESP8266WiFiSTA.cpp

```diff
@@ -381,17 +382,21 @@ bool ESP8266WiFiSTAClass::getAutoReconnect() {
 /**
  * Wait for WiFi connection to reach a result
  * returns the status reached or disconnect if STA is off
- * @return wl_status_t
+ * @return wl_status_t or -1 on timeout
  */
-uint8_t ESP8266WiFiSTAClass::waitForConnectResult() {
+int8_t ESP8266WiFiSTAClass::waitForConnectResult() {
     //1 and 3 have STA enabled
     if((wifi_get_opmode() & 1) == 0) {
         return WL_DISCONNECTED;
     }
-    while(status() == WL_DISCONNECTED) {
-        delay(100);
+    using esp8266::polledTimeout::oneShot;
+    oneShot timeout(60*1000); // number of milliseconds to wait before returning timeout error
+    while(!timeout) {
+        if(status() != WL_DISCONNECTED) {

```

Comment:
I forgot a yield() in my example :laughing: I suggest as first line inside the while.